File: newsfeedbackend/newsfeedbackend/src/main/kotlin/com/vdzon/newsfeedbackend/model/CategoryResult.kt
package com.vdzon.newsfeedbackend.model

data class CategoryResult(
    val categoryId: String,
    val categoryName: String,
    val articleCount: Int,
    val costUsd: Double,
    val searchResultCount: Int = 0,   // aantal resultaten van Tavily
    val filteredCount: Int = 0        // aantal over na datum-filter
)

File: frontend/lib/models/category_result.dart
class CategoryResult {
  final String categoryId;
  final String categoryName;
  final int articleCount;
  final double costUsd;
  final int searchResultCount;
  final int filteredCount;

  const CategoryResult({
    required this.categoryId,
    required this.categoryName,
    required this.articleCount,
    required this.costUsd,
    this.searchResultCount = 0,
    this.filteredCount = 0,
  });

  factory CategoryResult.fromJson(Map<String, dynamic> json) => CategoryResult(
        categoryId: json['categoryId'] as String,
        categoryName: json['categoryName'] as String,
        articleCount: json['articleCount'] as int? ?? 0,
        costUsd: (json['costUsd'] as num?)?.toDouble() ?? 0.0,
        searchResultCount: json['searchResultCount'] as int? ?? 0,
        filteredCount: json['filteredCount'] as int? ?? 0,
      );
}
